packages/sha1-browser/src/crossPlatformSha1.ts

```diff
@@ -1,14 +1,17 @@
 import { Sha1 as Ie11Sha1 } from "./ie11Sha1";
 import { Sha1 as WebCryptoSha1 } from "./webCryptoSha1";
-import { Hash, SourceData } from "@aws-sdk/types";
+import { Checksum, SourceData } from "@aws-sdk/types";
 import { supportsWebCrypto } from "@aws-crypto/supports-web-crypto";
 import { isMsWindow } from "@aws-crypto/ie11-detection";
 import { locateWindow } from "@aws-sdk/util-locate-window";
+import { convertToBuffer } from "@aws-crypto/util";
 
-export class Sha1 implements Hash {
-  private readonly hash: Hash;
+export class Sha1 implements Checksum {
+  private readonly secret?: SourceData;
+  private hash: Checksum;
 
   constructor(secret?: SourceData) {
+    this.secret = secret;
```

Comment:
Question:
It appears that we are introducing the `secret` field/property
just to facilitate `reset`, which is the same logic as `constructor`.

Can we introduce a private method that holds 
the common logic b/w `reset` and `constructor` such
that we do not need the `secret` field and do not repeat code?

(The answer maybe no... as `constructor` is clearly not void...)

Reply:
Yeah I tried this previously but got the following error:
`error TS2564: Property 'hash' has no initializer and is not definitely assigned in the constructor.`

Reply:
The `secret` field is required because `reset` method needs it to recreate the class fields such as `hash`.

I have tried adding a shared private method between `reset` and `constructor`, but unfortunately the `hash` needs to be defined explicitly within the constructor method. Otherwise we get the follow error:

`error TS2564: Property 'hash' has no initializer and is not definitely assigned in the constructor.`

Reply:
I have updated some of the constructors to call `reset` so there is no code duplication.

However, `secret` needs to be a class field because it is needed outside of constructor i.e. when `reset` is method